### PR TITLE
tessen: init at 2.1.2

### PR DESCRIPTION
**Pull request by magnouvean**
###### Description of changes
Added tessen package. Tessen is a wayland-equivelant to rofi-pass with some added features and more maintained. It also support other backends than rofi, for example bemenu. For more information see: https://github.com/ayushnix/tessen

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/tessen/default.nix
Comment (viraptor) [inline pkgs/applications/misc/tessen/default.nix:2]:
```suggestion
{ lib
, stdenv
```
The common style is to put `lib` first.

File: pkgs/applications/misc/tessen/default.nix
Comment (viraptor) [inline pkgs/applications/misc/tessen/default.nix:49]: The project seems to be using `gpl2plus` specifically. https://github.com/ayushnix/tessen/commit/f4294f1404212adc0784e54a7c6f448b19ff90d7

File: pkgs/applications/misc/tessen/default.nix
Comment (viraptor) [inline pkgs/applications/misc/tessen/default.nix:44]: The `wrapperPath` shouldn't be a part of the derivation set. Try this instead:
```suggestion
  fixupPhase = let
    wrapperPath = lib.makeBinPath [
      backend
      libnotify
      pass
      wl-clipboard
      wtype
    ];
  in ''
    patchShebangs $out/bin
    wrapProgram $out/bin/tessen --prefix PATH : ${wrapperPath}
  '';
```

**Review by viraptor — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/tessen/default.nix
Comment (viraptor) [inline pkgs/applications/misc/tessen/default.nix:48]: Sorry, only just noticed. Could you [capitalise the description](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes), please?

File: pkgs/applications/misc/tessen/default.nix
Comment (viraptor) [inline pkgs/applications/misc/tessen/default.nix:46]: Since this is linux/wayland-only, could you add `platform = platforms.linux`?

**Review by viraptor — COMMENTED** [review]
Awesome, looks good, just noticed some small issues.
Also, please squash the change - it should be a single commit only.

**Review by viraptor — APPROVED** [review]
As far as I can tell, it runs :)
I'm not a user of those tools, so I leave it to the maintainer to ensure it actually does the right thing.

**Comment by MonAaraj** [comment]
My PR at #187900 adds the same package but a more recent version. It also seems mine has a bit more progress in terms of reviews (though, it conflicts right now, but I'll make sure to solve it). Do I close mine? @magnouvean

**Comment by magnouvean** [comment]
> My PR at #187900 adds the same package but a more recent version. It also seems mine has a bit more progress in terms of reviews (furthermore, mine uses fetchFromSourcehut to fetch the sourcehut version which is the official repo, the github one is a mirror, i don't think that matters though). Do I close mine? @magnouvean 

That is fine. I think this PR is ready to merge as is, just waiting for more reviews. Just had a look at #187900 and seems good as well (though it seems bemenu is locked in as the backend, maybe this could be customizable?). If you think your PR will be merged before I am happy to close this though @MonAaraj :)

**Comment by MonAaraj** [comment]
I think we should see which one is preferred by reviewers enough to be merged first? I think the reviews on my PR have interesting code reviews that might apply to your PR too

File: pkgs/applications/misc/tessen/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/misc/tessen/default.nix:10]:
```suggestion
```

File: pkgs/applications/misc/tessen/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/misc/tessen/default.nix:44]:
```suggestion
  postFixup = ''
    wrapProgram $out/bin/tessen --prefix PATH : ${lib.makeBinPath [ rofi libnotify pass wl-clipboard wtype ]}
  '';
```
IIRC then fixupPhase should fix the shebangs already if we don't overwrite it.

File: pkgs/applications/misc/tessen/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/misc/tessen/default.nix:13]:
```suggestion
stdenvNoCC.mkDerivation rec {
```

**Comment by SuperSandro2000** [comment]
merged #187900 instead

**Pull request closed without merging**
